scripts/generate-graph-mermaid.py

```
@@ -0,0 +1,146 @@
+MAX_LABEL_LEN = 30
+TOP_N = 15
+MAX_EDGES = 30
+
+
+def truncate(label: str, max_len: int = MAX_LABEL_LEN) -> str:
+    if len(label) > max_len:
+        return label[: max_len - 1] + "…"
+    return label
+
+
+def main() -> None:
```

Ruff (0.15.12)

[warning] Too many branches (20 > 12)

(PLR0912)